### DISCUSSION TITLE
Month navigation dropdown and buttons in v8

### DISCUSSION THREAD

**Discussion by pptheberge** [gpbl/react-day-picker#1425] [Support] [answered]
Hi,
The option to have dropdowns builtin is a great addition in v8 as we no longer require custom code to do it. That being said, we're wondering if it's possible to have an option to display both the buttons and the dropdowns. We have a use case where in most cases the date to pick will be close so the buttons are the best way to navigate from a user experience perspective but the time where the date can be several months in the future is still non-negligible and asking the user to click repeatedly to get to the date is not an ideal experience.

Is there an easier way to achieve this than reimplementing the Caption from scratch? There doesn't seem to be a way to import YearsDropdown and MonthsDropdown.

Thanks in advance

**Comment by gpbl** [comment, marked as answer]
My problem was including _both_ as that there's a limited space:

<img width="314" alt="Screen Shot 2022-04-03 at 07 37 04" src="https://user-images.githubusercontent.com/120693/161428436-06fd9f42-9b78-4bc3-9eba-c49f427d3811.png">

Anyway, v8.0.2 has a [smaller Caption component](https://github.com/gpbl/react-day-picker/blob/master/packages/react-day-picker/src/components/Caption/Caption.tsx) and the new `CaptionNavigation` / `CaptionDropdowns` components that should help in your task.

**Comment by pptheberge** [comment]
Thank you. With your help, 8.0.2 was all that was required. A simple custom caption and slightly reducing the font size on the caption did the trick. 

Our code below for reference in case anyone else is trying to achieve this.

Thanks again,

```
export function CustomCaption(props) {
  const { classNames, styles } = useDayPicker();

  return (
    <div className={classNames.caption} style={styles.caption}>
        <CaptionDropdowns displayMonth={props.displayMonth} id={props.id} />
        <CaptionNavigation displayMonth={props.displayMonth} id={props.id} />
    </div>
  );
}
```

and when using DayPicker

....
 ```
     components={{
        Caption: CustomCaption,
        CaptionLabel: () => {
          return <></>;
        },
      }}
      styles={{
        caption: { fontSize: "90%" },
      }}
```
...
